### ISSUE TITLE
Issue with Azure OpenAI API usages with BERTopic

### ISSUE THREAD

**Issue by amitguptadumka** [MaartenGr/BERTopic#1263]
I have an issue working with custom GPT model with Bertopic.
So i had used  default GPT integrated model for topic clustering as per documentation and it worked like charm(code below):

openai.api_key ='sk-xxxxxxxxxxxx'
representation_model= OpenAI(model="gpt-3.5-turbo", delay_in_seconds=15, chat=True)
topic_model = BERTopic(representation_model=representation_model)
topics5, probs5 = topic_model.fit_transform(docs)
This was via my personal API key i had experimented on public data.

Now  i am trying to use Bertopic with Azure integrated OPENAI API key which i am trying for enterprise data. 
There are custom OpenAI GPT model deployed by enterprise and when i use it with a chat completion model its giving proper response. But when i am using this API key and custom model with the initial code(representation model) its throwing me error , i am not sure how to convert this chat completion code and use it to create topic cluster. i tried approaches with documentation but no luck.
The error im getting with this is: InvalidRequestError: Must provide an 'engine' or 'deployment_id' parameter to create a <class 'openai.api_resources.chat_completion.ChatCompletion'>
Basically the custom model is nothing but OpenAI GPT 3.5 turbo model.

**Comment by amitguptadumka** [comment]
Thanks. But as per this thread when i include custom model with 
from bertopic.representation import OpenAI
representation_model= OpenAI(engine="my-custom-embedding_model", delay_in_seconds=15, chat=True)

it does not accept "engine" as parameter.
TypeError: __init__() got an unexpected keyword argument 'engine'

**Comment by MaartenGr** [comment]
You can use the `generator_kwargs` in the `OpenAI` module to specify additional kwargs for the ` .create` function:

```python
from bertopic.representation import OpenAI
representation_model= OpenAI(generator_kwargs={"engine":"my-custom-embedding_model")
```

**Comment by amitguptadumka** [comment]
Thanks @MaartenGr . This worked like charm.

**Comment by ninavdPipple** [comment]
I don't know if others encountered the same problem, but when I pass a 'model' in the generator_kwargs, I get an error that a certain arugment is passed twice into the function. If I look at the code, this actually makes sense as first it is unpacked in self.model and eventually both self.model and generator_kwargs are passed into the completions. I fixed this by changing 

        if self.generator_kwargs.get("model"):
            self.model = generator_kwargs.get("model")

to 

        if self.generator_kwargs.get("model"):
            self.model = generator_kwargs.get("model")
            del self.generator_kwargs["model"]

Maybe this is just my particular case where this is happening, but if not this might be a potential solution to others running in trouble.

Upon inspecting the code a bit further I also figured you can simply pass it in the model argument of OpenAI such that there is no need to pass it in the generator_kwargs and this problem is also avoided.

**Comment by MaartenGr** [comment]
@ninavdPipple That indeed seems to be a bug! Adding that line would definitely solve the problem. If you want, a PR would be greatly appreciated!
